test/controllers/desks_controller_test.rb

```diff
@@ -44,12 +45,14 @@ class DesksControllerTest < ActionDispatch::IntegrationTest
   end
 
   test "#update updates desk and responds with updated desk" do
-    put desk_path(@desk, format: :json), params: { desk: { occupied: !@desk.occupied } }
+    @desk.occupied = !@desk.occupied
```

Comment:
This is dangerous, we are changing the database entry in the test driver although we actually want the controller to change it. In this case it probably doesn't matter because we don't save it but it could easily result in unexpected behavior.

I think a preferable approach would be to pass the opposite "occupied" state as a parameter in the `put` and then to check that the underlying database entry has also been changed.

Changes:
1. remove `@desk.occupied = !@desk.occupied`
2. store original state: `occupied = @desk.occupied`
2. `put desk_path(@desk, format: :json), params: { desk: { occupied: !occupied } }`
3. add after put: `@desk.reload`
3. check that db entry has changed: `assert_equal !occupied, @desk.occupied`